refactor: remove the expiry field

divyanshugit opened the pull request:
### Description

<!--- Include a summary of the change and relevant motivation/context. List any dependencies that are required for this change. --->

Removed the `expiry` field from login response model

Fixes #679

### Type of Change:

- Code


### Checklist:

<!-- **Delete irrelevant options.** -->

- [x] My PR follows the style guidelines of this project
- [x] Removed references to expiry timestamp field from POST /login

**Code/Quality Assurance Only**

- [x] New and existing unit tests pass locally with my changes

divyanshugit committed:
removed the expiry field

divyanshugit committed:
Removed the expiry timestamp

isabelcosta commented on app/api/models/user.py:
```diff
@@ -151,37 +151,6 @@ def add_models_to_namespace(api_namespace):
-    "Login response data model",
-    {
-        "access_token": fields.String(required=True, description="User's access token"),
-        "expiry": fields.Float(
-            required=True, description="Access token expiry UNIX timestamp"
-        ),
```
these are the only lines that should be removed in this pull request :) All the rest you removed, you should keep

divyanshugit replied:
Thanks for your review :)

isabelcosta reviewed (changes requested):
@divyanshugit You are almost there :) You just have to remove 3 lines of code, after that it the work should be complete.

divyanshugit committed:
Removed expiry timestamp

divyanshugit committed:
Update user.py

Stored to previous version of file

divyanshugit committed:
Update messages.py

Stored to previous version of file

divyanshugit committed:
Update test_api_login.py

Stored to previous version of file

divyanshugit committed:
Update user.py

Stored to previous version of file

divyanshugit committed:
Update test_api_refresh.py

Stored to previous version of file

isabelcosta commented:
@divyanshugit can you please only commit what you have done in this PR? From what I understand your code should only include 3 lines of code. A recommendation we have in our "Contributions Guidelines" is to create a branch specific to an issue you are solving. I think you are accumulating commits you did on a previous PR. The problem is that we squash commits, so the commit that goes to `develop` branch is different than the ones you have in your `develop` branch from previous PR.

divyanshugit commented:
@isabelcosta  Sorry, I didn't saw that It was just updated before my PR. I just copied the code from mentorship-backend/app/api/resources/user.py / . Again, Sorry for my mistake I will make sure that it will not happen again in future.

divyanshugit committed:
Update user.py

divyanshugit committed:
Update user.py

divyanshugit committed:
Update user.py

divyanshugit committed:
Update user.py

divyanshugit committed:
Update user.py

divyanshugit commented:
@isabelcosta  Is this okay ? Can you please suggest me some new issues on which I can start working !

divyanshugit committed:
Update user.py

divyanshugit committed:
Update user.py

Removed the expiry field

vj-codes reviewed (commented):
@divyanshugit A good implementation would be to checkout to a suitable branch and then send a PR when you will start working on other issues!

isabelcosta reviewed (dismissed):
Great work @divyanshugit ! Thank you for your contribution! Now I just need one little test to see if login is working as expected :)

isabelcosta commented:
@anitab-org/qa-team to test this, you can see if Login API works well as before, that should be enough :)

SanketDG reviewed (changes requested):
There is a `TODO` on https://github.com/anitab-org/mentorship-backend/pull/704/files#diff-a831ce61f9c82aab25c2f42e88908329R155, we may also want to remove that.

Good to merge after that!

isabelcosta commented:
@SanketDG I don't see such TODO, the current version of the code that @divyanshugit changed  does not have TODO

SanketDG commented:
@isabelcosta I still see a TODO on line 155 of the changed file

isabelcosta commented:
@SanketDG very nice catch! At first viewing I did not see it, I had to click to expand, your change request makes sense!

![image](https://user-images.githubusercontent.com/11148726/90075069-728a0400-dcf4-11ea-983d-c90cfb73a41e.png)

@divyanshugit can you please remove the line 155?

divyanshugit committed:
refactor: remove the expiry field

Removed comment !

SanketDG reviewed (approved):
Looks good :rocket:

isabelcosta reviewed (dismissed):
Great work @divyanshugit 🎉

SanketDG reviewed (dismissed):
Nicely done! @divyanshugit

isabelcosta commented:
@anitab-org/qa-team can you please test this PR?

divyanshugit committed:
fix: refactor HTTP error codes(#632)

Now, If the task was already achieved, then Conflict is returned instead of Bad Request.

divyanshugit committed:
Revert "fix: refactor HTTP error codes(#632)"

This reverts commit 20eaf7eda3d4516a3cb260656628c666982debc3.

SanketDG reviewed (approved):
Ready for Merge :tada: 

@divyanshugit Tip: You do not need to revert your own commit in your own PR! :P

divyanshugit commented:
@SanketDG I mistakenly commit in the wrong branch. I was solving the issue #632 in that commit. Sorry for that mistake.

foongminwong reviewed (approved):
The changes made in this PR were tested locally. Following are the results:

1. Code review - Done

2. All possible responses were tested as below:

   * _Before PR 704_  
    _Screenshot/gif/url_:  

    
       ![before-ms-backend-pr-704](https://user-images.githubusercontent.com/23441795/91561191-bffda680-e96d-11ea-8914-8dd16cdd901c.gif)

    _Expected Result_:  As a user, I can view the `expiry` timestamp on MS Swagger UI.
    _Actual Result_: Same as expected.

   * _After PR 704_  
    _Screenshot/gif/url_:  

  
     ![after-ms-backend-pr-704](https://user-images.githubusercontent.com/23441795/91561210-c5f38780-e96d-11ea-9a66-2349368535c0.gif)


    _Expected Result_:  The `expiry` timestamp on MS Swagger UI is removed. As a user, I shouldn't see it on MS Swagger UI.
    _Actual Result_: Same as expected.

3. Comments: Good work @divyanshugit !!
 
4. OS Version: Windows 10

isabelcosta commented:
@foongminwong your test report is so amazing 👏
